qiskit/circuit/library/standard_gates/ecr.py

```diff
@@ -112,3 +112,6 @@ def _define(self):
     def inverse(self):
         """Return inverse ECR gate (itself)."""
         return ECRGate()  # self-inverse
+
+    def __eq__(self, other):
+        return isinstance(other, ECRGate)
```

Comment:
If someone were to extend `ECRGate` via inheritance, this definition of `__eq__` would be wrong for that subclass. Is it worth instead defining this once for `SingletonGate` as:

```py
def __eq__(self, other):
    return isinstance(other, type(self))
```

Reply:
That wouldn't actually be correct for singleton gates, where the singleton instance is a separate part of the inheritance tree to the non-singleton versions.

I'm not super convinced that `__eq__` should attempt to drive additional logic within subclasses; we can't really know what additional information a subclass might add on, nor what should really cause them to be considered equal.  It largely seems to me that subclasses should almost always be responsible for overriding `__eq__`.

Reply:
Ah, I forgot about that nuance with the class hierarchy. It seems to me like `__eq__` should behave the same for all `SingletonGate`s. If there is more state to compare, then I'd think it's no longer a `SingletonGate` from a traiting perspective.